azure/microservice-ecommerce/terraform/variables.tf.tmpl

```diff
@@ -0,0 +1,62 @@
+variable "resource_group" {
+  # * ResourceGroup
+  description = "A resource group is a collection of resources that share the same lifecycle, permissions, and policies"
+}
+
+variable "cluster_name" {
+  # * ClusterName
+  description = "The name of the Azure Kubernetes Service cluster"
+}
+
+variable "location" {
+  # * AKSRegion
+  description = "Location of resources"
+}
+
+variable "vm_size" {
+  description = "The type of VM"
+  default     = "Standard_B2ms"
+}
+
+variable "node_count" {
+  description = "The size of the virtual machines that will form the nodes in the cluster. This cannot be changed after creating the cluster"
+  default     = 3
+}
+
+variable "admin_username" {
+  # * AdminUsername
+  description = "The username for the cluster (cannot be 'admin')"
+  default     = "jjohnson"
+}
+
+variable "ssh_public_key_path" {
+  # * AdminUsernameSSHKeyPath
+  description = ""
+  default     = "{{.AdminUsernameSSHKeyPath | trim}}"
```

Comment:
Done

Reply:
Problem. Passing this key as an argument to `terraform apply` causes the command to fail because a public key usually contains two spaces. It's these extra space that confuse Terraform

Reply:
I can try adding escaped double quotes and see if that fixes the issue